src/phoenix/trace/openai/instrumentor.py

```diff
@@ -1,683 +1,43 @@
-import json
-from collections import defaultdict
-from dataclasses import dataclass, field
-from datetime import datetime, timezone
-from enum import Enum
-from inspect import BoundArguments, signature
-from types import TracebackType
-from typing import (
-    Any,
-    AsyncIterator,
-    Callable,
-    ContextManager,
-    Coroutine,
-    DefaultDict,
-    Dict,
-    Iterator,
-    List,
-    Mapping,
-    Optional,
-    Tuple,
-    Type,
-    TypeVar,
-    cast,
-)
-
-import openai
-from openai import AsyncStream, Stream
-from openai.types.chat import ChatCompletion, ChatCompletionChunk
-from typing_extensions import ParamSpec
-from wrapt import ObjectProxy
+from typing import Optional
 
-from phoenix.trace.schemas import (
-    MimeType,
-    SpanAttributes,
-    SpanEvent,
-    SpanException,
-    SpanKind,
-    SpanStatusCode,
-)
-from phoenix.trace.semantic_conventions import (
-    INPUT_MIME_TYPE,
-    INPUT_VALUE,
-    LLM_FUNCTION_CALL,
-    LLM_INPUT_MESSAGES,
-    LLM_INVOCATION_PARAMETERS,
-    LLM_OUTPUT_MESSAGES,
-    LLM_TOKEN_COUNT_COMPLETION,
-    LLM_TOKEN_COUNT_PROMPT,
-    LLM_TOKEN_COUNT_TOTAL,
-    MESSAGE_CONTENT,
-    MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON,
-    MESSAGE_FUNCTION_CALL_NAME,
-    MESSAGE_NAME,
-    MESSAGE_ROLE,
-    MESSAGE_TOOL_CALLS,
-    OUTPUT_MIME_TYPE,
-    OUTPUT_VALUE,
-    TOOL_CALL_FUNCTION_ARGUMENTS_JSON,
-    TOOL_CALL_FUNCTION_NAME,
+from openinference.instrumentation.openai import (
+    OpenAIInstrumentor as OpenInferenceOpenAIInstrumentor,
 )
-from phoenix.trace.utils import get_stacktrace
-
-from ..tracer import Tracer
-
-ParameterSpec = ParamSpec("ParameterSpec")
-GenericType = TypeVar("GenericType")
-AsyncCallable = Callable[ParameterSpec, Coroutine[Any, Any, GenericType]]
-Parameters = Mapping[str, Any]
-OpenInferenceMessage = Dict[str, str]
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk import trace as trace_sdk
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor
 
-INSTRUMENTED_ATTRIBUTE_NAME = "is_instrumented_with_openinference_tracer"
-
-
-class RequestType(Enum):
-    CHAT_COMPLETION = "chat_completion"
-    COMPLETION = "completion"
-    EMBEDDING = "embedding"
+from phoenix.trace.exporter import _OpenInferenceExporter
+from phoenix.trace.tracer import Tracer
```

Comment:
suggestion: maybe we should start eliminating the dependency on the defunct class.
```suggestion
```